cicd-scripts/app_install.sh

```diff
@@ -1,51 +1,68 @@
 #!/bin/bash
 
+
+# A hack to get the environment running without ansible local env variables
+# This block of code will eventually be removed once ansible task is completed
+SPIDER_PYTHON_VERSION=3.12
+SPIDER_STAGING_URLS_API=https://staging.search.usa.gov/urls
+spider_local_path=/etc/profile.d/spider_local.sh
+
+# Writing environment variables to the profile file
+echo "
+export SPIDER_PYTHON_VERSION=${SPIDER_PYTHON_VERSION}
+export SPIDER_STAGING_URLS_API=${SPIDER_STAGING_URLS_API}
+" | tee "$spider_local_path" > /dev/null
+
+# Source the script to update the current shell's environment
+source "$spider_local_path"
+### TODO: Remove the above code block after ansible is fully implmented
+
+
 # Update and upgrade the system without prompting for confirmation
 sudo apt-get update -y
 sudo apt-get upgrade -y
 
 # Install necessary system dependencies
 sudo apt-get install -y python-setuptools python-pip
 
-# Function to install Python 3.12
 install_python() {
-    echo "Installing Python 3.12..."
+    echo "Installing ${SPIDER_PYTHON_VERSION}"
     sudo apt-get install -y build-essential checkinstall libreadline-dev \
                             libncursesw5-dev libssl-dev libsqlite3-dev \
                             tk-dev libgdbm-dev libc6-dev libbz2-dev \
                             zlib1g-dev openssl libffi-dev
 
-    # Download Python 3.12 source code
+    # Download Python source code
     cd /usr/src
-    sudo wget https://www.python.org/ftp/python/3.12.0/Python-3.12.0.tgz
-    sudo tar xzf Python-3.12.0.tgz
+    sudo wget https://www.python.org/ftp/python/${SPIDER_PYTHON_VERSION}.0/Python-${SPIDER_PYTHON_VERSION}.0.tgz
+    sudo tar xzf Python-${SPIDER_PYTHON_VERSION}.0.tgz
 
-    # Build and install Python 3.12
-    cd Python-3.12.0
+    # Build and install Python
+    cd Python-${SPIDER_PYTHON_VERSION}.0
     sudo ./configure --enable-optimizations
     sudo make altinstall
 
-    echo "Python 3.12 has been installed."
+    echo "Python ${SPIDER_PYTHON_VERSION} has been installed."
 }
 
-# Check if Python 3.12 is installed
-if command -v python3.12 &>/dev/null; then
-    echo "Python 3.12 is already installed: $(python3.12 --version)"
+# Check if Python is installed
+if command -v python${SPIDER_PYTHON_VERSION} &>/dev/null; then
+    echo "Python ${SPIDER_PYTHON_VERSION} is already installed: $(python${SPIDER_PYTHON_VERSION} --version)"
 else
-    echo "Python 3.12 is not installed. Installing Python 3.12..."
+    echo "Python ${SPIDER_PYTHON_VERSION} is not installed. Installing Python ${SPIDER_PYTHON_VERSION}..."
     install_python
 fi
 
-# Install virtualenv using Python 3.12's pip
-sudo /usr/local/bin/python3.12 -m pip install --upgrade pip
-sudo /usr/local/bin/python3.12 -m pip install virtualenv
+# Install virtualenv using Python pip
+sudo /usr/local/bin/python${SPIDER_PYTHON_VERSION} -m pip install --upgrade pip
+sudo /usr/local/bin/python${SPIDER_PYTHON_VERSION} -m pip install virtualenv
```

Comment:
i don't see where we are using virtualenv.  Below you are using venv the built in module.  I don't think we need this installed.

Reply:
Its aliased as `venv`. It might already be installed (usually it's part of python), but it's not guaranteed

Reply:
Did you alias it somewhere or is that built in functionality? venv is included in the standard python library.  

https://docs.python.org/3.12/library/venv.html
vs
https://pypi.org/project/virtualenv/

and from https://virtualenv.pypa.io/en/latest/:

> virtualenv is a tool to create isolated Python environments. Since Python 3.3, a subset of it has been integrated into the standard library under the [venv module](https://docs.python.org/3/library/venv.html). The venv module does not offer all features of this library, to name just a few more prominent.

Reply:
Ah you're right. I always thought they were the same thing and that we can install python without it and then add it via virtualenv